### ISSUE TITLE
conky screws up, when system time makes big steps.

### ISSUE THREAD

**Issue by beta-tester** [brndnmtthws/conky#942]
**Issue**

i have a Raspberry Pi, that does not have an internal battery buffered RTC clock.
so it happens, that first the fake-hwclock takes the time of the last shutdown and receiving the correct time will take some time (from internet, GPS or other external sources).
when conky is already running and the system time makes a bigger step, conky will screw up...
e.g.: its transparent background will turn solid black and multiple elements will be caotically overwritten.
restarting `sudo killall -SIGUSR1 conky` will bring the display back to normal.

**Comment by beta-tester** [comment]
![Screenshot from 2020-02-13 16-02-13](https://user-images.githubusercontent.com/4750719/74447877-52c76f80-4e7a-11ea-8488-66a03a40afe9.png)![Screenshot from 2020-02-13 16-01-23](https://user-images.githubusercontent.com/4750719/74447894-5a871400-4e7a-11ea-8828-985bb3be8481.png)

**Comment by bi4k8** [comment]
It seems likely that there's something wrong with Conky's logic around its update interval and which clock (monotonic, etc.) it relies on. This might be a bit of a pain to write a test case for, but also seems worth fixing.

**Comment by beta-tester** [comment]
i am not sure if this was the configuration i used back then:
https://gist.github.com/beta-tester/efa3b636703f0a61a0be7a2a2623f97f

**Comment by Caellian** [comment]
The buffer seems to not get cleared, which is weird given that the [code related to clearing X11 buffer](https://github.com/brndnmtthws/conky/blob/2ffc5d67a67d5aeb43f9b7b98180e49c9fb2eb53/src/display-x11.cc#L652) doesn't depend on time - i.e. it should happen before every render.

In `main_loop` we pass "time until next update" while trying to select display buffer file [here](https://github.com/brndnmtthws/conky/blob/2ffc5d67a67d5aeb43f9b7b98180e49c9fb2eb53/src/conky.cc#L1739) as timeout value.
- This would make conky not update the content altogether, instead of keeping the stale buffer.

And the other (last) place in the `main_loop` is [here](https://github.com/brndnmtthws/conky/blob/2ffc5d67a67d5aeb43f9b7b98180e49c9fb2eb53/src/conky.cc#L1748), where we "wait before next draw" in order to respect redraw interval.

The only other thing that comes to mind is X11 discarding

**Comment by Caellian** [comment]
I self assigned this issue because I'd like to add eager rendering (no delay) as well. Someone else can take over this issue if they want to and I didn't open a PR in the meantime.

**Comment by Caellian** [comment]
Marked with low priority as it's a very niche issue. Unassigned myself because it's hard to tell when I'll have time to deal with it and it's less pressing than Wayland support.

Some additional thoughts: maybe this has to do with XDBE code and not even conky? It's hard to tell at this point because this issue is relatively old, OP probably no longer has the same setup around.

I suggest letting this issue be closed if it gets stale again. I can reopen it someone has the same problem as OP did 4 years ago.
